### ISSUE TITLE
In the view hierarchy (XML) of a page, the TextField widget does not show the content description.

### ISSUE THREAD

**Issue by aislan11110** [flutter/flutter#153495]
### Steps to reproduce

1.  create a Semantics for the TextField
2.  capture the view hierarchy of an Android device screen and export it to an XML file using UIAutomator.
3.  When inspecting this XML, notice that the node referring to the TextField, the content-desc, is empty.

### Expected results

```
<node bounds="[0,1033][1080,1218]" checkable="false" checked="false"
                                        class="android.widget.EditText"
                                        clickable="false" content-desc="LoginInputEmail"   
                                        enabled="true" focusable="true" focused="false"
                                        index="1" long-clickable="false" package="com.example.test_orthodox"
                                        password="false" resource-id="" scrollable="false"
                                        selected="false" text="">
                                       </node>
```

### Actual results

```
<node bounds="[0,1033][1080,1218]" checkable="false" checked="false"
                                        class="android.widget.EditText"
                                        clickable="false" content-desc=""
                                        enabled="true" focusable="true" focused="false"
                                        index="1" long-clickable="false" package="com.example.test_orthodox"
                                        password="false" resource-id="" scrollable="false"
                                        selected="false" text="">
                                        <node NAF="true" bounds="[0,1033][1080,1218]" checkable="false" checked="false"
                                            class="android.widget.EditText"
                                            clickable="true" content-desc=""
                                            enabled="true" focusable="true" focused="false"
                                            index="0" long-clickable="false" package="com.example.test_orthodox"
                                            password="false" resource-id=""
                                            scrollable="false" selected="false"
                                            text="" />
                                    </node>
```

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Semantics(
              label: 'LoginInputEmail',
              textField: true,
              child: TextField(
                decoration: InputDecoration(
                  labelText: 'Email',
                  border: OutlineInputBorder(),
                ),
              ),
            ),
            const SizedBox(height: 20),
            Semantics(
              label: "XXXXXXXXXXXXXXXXXXXXXXXXXx",
              button: true,
              child: ElevatedButton(
                onPressed: () {
                  //no action
                },
                child: const Text("login"),
              ),
            )
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/user-attachments/assets/21687788-4204-439d-a046-1eba308e4ed3)
![image](https://github.com/user-attachments/assets/20f5fe0b-50c0-45b8-823a-3dc147a29052)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel main, 3.24.0-1.0.pre.592, on Microsoft Windows [versÆo 10.0.22631.4037], locale pt-BR)
    • Flutter version 3.24.0-1.0.pre.592 on channel main at C:\Users\aisla\dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 99f00a1309 (6 hours ago), 2024-08-15 03:00:25 -0400
    • Engine revision 971ddd9fe1
    • Dart version 3.6.0 (build 3.6.0-146.0.dev)
    • DevTools version 2.38.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\aisla\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio2\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0--11852314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[!] Android Studio (version 2023.2)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to determine bundled Java version.
    • Try updating or re-installing Android Studio.

[√] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio2
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0--11852314)

[√] Connected device (4 available)
    • SM A546E (mobile) • RXCW50EDGFR • android-arm64  • Android 14 (API 34)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [versÆo 10.0.22631.4037]
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 127.0.6533.101
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 127.0.2651.98

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

```

</details>

**Comment by huycozy** [comment]
Thanks for the report. I also can see that `content-desc` in TextField is missing via UiAutomator and Appium tool.

UiAutomator command to dump the view hierarchy:
```console
➜  repro_issues adb exec-out uiautomator dump
UI hierchary dumped to: /sdcard/window_dump.xml
```

<details>
<summary>flutter doctor (stable and master)</summary>

```console
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.6 23G80 darwin-x64, locale en-VN)
    • Flutter version 3.24.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (7 days ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2
```

```console
[!] Flutter (Channel master, 3.24.0-1.0.pre.605, on macOS 14.6 23G80 darwin-x64, locale en-VN)
    • Flutter version 3.24.0-1.0.pre.605 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bced008679 (26 minutes ago), 2024-08-15 22:29:24 -0400
    • Engine revision a8fefc8118
    • Dart version 3.6.0 (build 3.6.0-146.0.dev)
    • DevTools version 2.38.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.
```
</details>

**Comment by justinmc** [comment]
@chunhtai Looks like something for the new `team-accessibility`! But let me know if I'm wrong.

**Comment by chunhtai** [comment]
This is WAI, the textfield will set Text and hintText but not contentdescription. I believe this isn't a way to current by hintText in UiAutomator yet. I don't think there is anything we can do other than asking UiAutomator to be able to dump text and hinttext.
